src/test/java/de/tum/in/www1/artemis/TextClusterResourceIntegrationTest.java

```diff
@@ -0,0 +1,190 @@
+import de.tum.in.www1.artemis.web.rest.TextClusterResource;
+
+public class TextClusterResourceIntegrationTest extends AbstractSpringIntegrationBambooBitbucketJiraTest {
+
+    @Autowired
+    private FeedbackRepository feedbackRepository;
+
+    @Autowired
+    private TextSubmissionRepository textSubmissionRepository;
+
+    @Autowired
+    private StudentParticipationRepository studentParticipationRepository;
+
+    @Autowired
+    private TextClusterResource textClusterResource;
+
+    @Autowired
+    private TextClusterRepository textClusterRepository;
+
+    @Autowired
+    private TextExerciseUtilService textExerciseUtilService;
+
+    @Autowired
+    private TextBlockRepository textBlockRepository;
+
+    @Autowired
+    private ResultRepository resultRepository;
+
+    private Exercise exercise;
+
+    /**
+     * Initializes the database with a course that contains a tutor and a text submission
+     */
+    @BeforeEach
+    public void initTestCase() throws Error {
+        Course course = database.createCourseWithInstructorAndTextExercise("instructor1");
+
+        exercise = course.getExercises().iterator().next();
+        StudentParticipation studentParticipation = studentParticipationRepository.findAll().get(0);
+
+        TextSubmission textSubmission = ModelFactory.generateTextSubmission("This is Part 1, and this is Part 2. There is also Part 3.", Language.ENGLISH, true);
+        textSubmissionRepository.save(textSubmission);
+
+        database.addResultToSubmission(textSubmission, AssessmentType.AUTOMATIC);
+
+        Set<TextBlock> textBlocks = Set.of(new TextBlock().text("This is Part 1,").startIndex(0).endIndex(15).automatic(),
+                new TextBlock().text("and this is Part 2.").startIndex(16).endIndex(35).automatic(),
+                new TextBlock().text("There is also Part 3.").startIndex(36).endIndex(57).automatic());
+        int[] clusterSizes = { 3 };
+        List<TextCluster> clusters = textExerciseUtilService.addTextBlocksToCluster(textBlocks, clusterSizes, (TextExercise) exercise);
+
+        clusters.get(0).setDisabled(true);
+        textBlocks.forEach(block -> {
+            block.setSubmission(textSubmission);
+            block.setCluster(clusters.get(0));
+            block.computeId();
+        });
+
+        textSubmission.setParticipation(studentParticipation);
+        textSubmissionRepository.save(textSubmission);
+        textClusterRepository.saveAll(clusters);
+        textBlockRepository.saveAll(textBlocks);
+    }
+
+    @AfterEach
+    public void resetDatabase() {
+        database.resetDatabase();
+    }
+
+    /**
+    * Checks the response data from retrieving cluster statistics is returned properly
+    */
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void testGetClusterStats_forAllValuesSet() {
+        ResponseEntity<List<TextClusterRepository.TextClusterStats>> responseEntity = textClusterResource.getClusterStats(exercise.getId());
```

Comment:
*Applies to the whole test class:*
We usually do the integration tests differently, by calling the REST endpoint and not the method of the resource.

We use the inherited `RequestUtilService request` like so:
```java
List<Notification> notifications = request.getList("/api/notifications", HttpStatus.OK, Notification.class);
```

It might be a good idea to make `TextClusterStats` a DTO or object in the domain package then.

Reply:
Switched all of them to `request` instead 👍